cupy/core/core.pyx

```diff
@@ -1091,7 +1091,7 @@ cdef class ndarray:
         if not isinstance(y, ndarray) and _should_use_rop(x, y):
             return NotImplemented
         else:
-            return _linalg.matmul(x, y)
+            return cupy.linalg._product.matmul(x, y)
```

Comment:
Silly question: can't we wrap `_linalg._matmul()` with gufunc only when dispatching is required (i.e. inside `__array_ufunc__`)?

Reply:
btw the comment above (line 1090) can be removed.

Reply:
If we only wrap on dispatching we lose the ability of supporting axis and the other gufunc keywords in the regular matmul.
Also it would pollute the dispatcher by needing to check if the function is matmul or not.

Reply:
Thanks, Emilio!